### ISSUE TITLE
"My Account" menu doesn't allow user to tab backwards from "Sign out" button

### ISSUE THREAD

**Issue by akegan** [bloom-housing/bloom#2242]
**Describe the bug**
The My Account menu allows a user to tab forward through the content when it's open, but if you tab to the "sign out" button, when you try to SHIFT + tab back to get to the previous item, the menu just closes.

**To Reproduce**
1. Go to [the menu dropdown component](https://storybook.bloom.exygy.dev/?path=/story/headers-site-header--with-menu-dropdowns) on storybook dev
2. Tab through to get focus on "my account"
3. Hit enter to open the menu
4. Tab down to "sign out"
5. Hit shift + tab to try to go back to the previous menu item

**Expected behavior**
Should be able to go back from "Sign out" to "account settings"

**Screenshots**
![image](https://user-images.githubusercontent.com/11825994/144143355-59a0e8dc-e47c-4066-bd1f-ce78a5cce560.png)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: Macbook pro
 - OS: Mac big sur
 - Browser Chrome 
 - Version 96

**Additional context**
It looks like it's related to the fact that some of the items in the menu are links, and the sign-out link is secretly a button

**Comment by emilyjablonski** [comment]
Fixed with Seeds
